.github/workflows/build-os-image.yml

```diff
@@ -912,8 +914,32 @@ jobs:
             echo -e "\`\`\`"
           } >> "$GITHUB_STEP_SUMMARY"
 
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: ${{ inputs.ref || github.head_ref }}
+
+      - name: Create CLI compatibility information artifact
+        shell: bash
+        run: |
+          go run ./hack/cli-k8s-compatibility/main.go \
+            --ref=${{  needs.build-settings.outputs.ref }} \
+            --stream=${{  needs.build-settings.outputs.stream }} \
+            --version=${{  needs.build-settings.outputs.imageVersion }} \
+            > info.json
+
+      - name: Upload CLI compatibility information to S3
+        shell: bash
+        run: |
+          aws s3 cp \
+            "info.json" \
+            "s3://cdn-constellation-backend/${{ needs.build-settings.outputs.cliApiBasePath }}/info.json" \
+            --no-progress
+          echo -e "[CLI compatibility information](https://cdn.confidential.cloud/${{ needs.build-settings.outputs.cliApiBasePath }}/info.json)" >> "$GITHUB_STEP_SUMMARY"
```

Comment:
You can move this step into the Go hack tool, too. Use the versionsapi/client update function to upload.

Reply:
see our teams conv regarding this